// react-native-paper-dates/lib/commonjs/Date/DayName.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _reactNative = require("react-native");
var _reactNativePaper = require("react-native-paper");
var _jsxRuntime = require("react/jsx-runtime");
function DayName({
  label
}) {
  const theme = (0, _reactNativePaper.useTheme)();
  let textFont = theme?.isV3 ? theme.fonts.bodySmall : theme.fonts.medium;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
    style: styles.dayName,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativePaper.Text, {
      maxFontSizeMultiplier: 1.5,
      style: [styles.dayNameLabel, {
        ...textFont,
        color: theme.colors.onSurface
      }],
      selectable: false,
      children: label
    })
  });
}
const styles = _reactNative.StyleSheet.create({
  dayName: {
    alignItems: 'center',
    flex: 1
  },
  dayNameLabel: {
    fontSize: 14,
    opacity: 0.7
  }
});
var _default = exports.default = /*#__PURE__*/(0, _react.memo)(DayName);
