// react-native-paper-dates/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Calendar", {
  enumerable: true,
  get: function () {
    return _Calendar.default;
  }
});
Object.defineProperty(exports, "DatePickerInput", {
  enumerable: true,
  get: function () {
    return _DatePickerInput.default;
  }
});
Object.defineProperty(exports, "DatePickerModal", {
  enumerable: true,
  get: function () {
    return _DatePickerModal.default;
  }
});
Object.defineProperty(exports, "DatePickerModalContent", {
  enumerable: true,
  get: function () {
    return _DatePickerModalContent.default;
  }
});
Object.defineProperty(exports, "TimePicker", {
  enumerable: true,
  get: function () {
    return _TimePicker.default;
  }
});
Object.defineProperty(exports, "TimePickerModal", {
  enumerable: true,
  get: function () {
    return _TimePickerModal.default;
  }
});
Object.defineProperty(exports, "ar", {
  enumerable: true,
  get: function () {
    return _ar.default;
  }
});
Object.defineProperty(exports, "ca", {
  enumerable: true,
  get: function () {
    return _ca.default;
  }
});
Object.defineProperty(exports, "cs", {
  enumerable: true,
  get: function () {
    return _cs.default;
  }
});
Object.defineProperty(exports, "da", {
  enumerable: true,
  get: function () {
    return _da.default;
  }
});
Object.defineProperty(exports, "de", {
  enumerable: true,
  get: function () {
    return _de.default;
  }
});
Object.defineProperty(exports, "el", {
  enumerable: true,
  get: function () {
    return _el.default;
  }
});
Object.defineProperty(exports, "en", {
  enumerable: true,
  get: function () {
    return _en.default;
  }
});
Object.defineProperty(exports, "enGB", {
  enumerable: true,
  get: function () {
    return _enGB.default;
  }
});
Object.defineProperty(exports, "es", {
  enumerable: true,
  get: function () {
    return _es.default;
  }
});
Object.defineProperty(exports, "fi", {
  enumerable: true,
  get: function () {
    return _fi.default;
  }
});
Object.defineProperty(exports, "fr", {
  enumerable: true,
  get: function () {
    return _fr.default;
  }
});
Object.defineProperty(exports, "getTranslation", {
  enumerable: true,
  get: function () {
    return _utils.getTranslation;
  }
});
Object.defineProperty(exports, "he", {
  enumerable: true,
  get: function () {
    return _he.default;
  }
});
Object.defineProperty(exports, "hi", {
  enumerable: true,
  get: function () {
    return _hi.default;
  }
});
Object.defineProperty(exports, "id", {
  enumerable: true,
  get: function () {
    return _id.default;
  }
});
Object.defineProperty(exports, "it", {
  enumerable: true,
  get: function () {
    return _it.default;
  }
});
Object.defineProperty(exports, "ja", {
  enumerable: true,
  get: function () {
    return _ja.default;
  }
});
Object.defineProperty(exports, "ko", {
  enumerable: true,
  get: function () {
    return _ko.default;
  }
});
Object.defineProperty(exports, "nl", {
  enumerable: true,
  get: function () {
    return _nl.default;
  }
});
Object.defineProperty(exports, "noNO", {
  enumerable: true,
  get: function () {
    return _noNO.default;
  }
});
Object.defineProperty(exports, "pl", {
  enumerable: true,
  get: function () {
    return _pl.default;
  }
});
Object.defineProperty(exports, "pt", {
  enumerable: true,
  get: function () {
    return _pt.default;
  }
});
Object.defineProperty(exports, "registerTranslation", {
  enumerable: true,
  get: function () {
    return _utils.registerTranslation;
  }
});
Object.defineProperty(exports, "ro", {
  enumerable: true,
  get: function () {
    return _ro.default;
  }
});
Object.defineProperty(exports, "ru", {
  enumerable: true,
  get: function () {
    return _ru.default;
  }
});
Object.defineProperty(exports, "sv", {
  enumerable: true,
  get: function () {
    return _sv.default;
  }
});
Object.defineProperty(exports, "th", {
  enumerable: true,
  get: function () {
    return _th.default;
  }
});
Object.defineProperty(exports, "tr", {
  enumerable: true,
  get: function () {
    return _tr.default;
  }
});
Object.defineProperty(exports, "ukUA", {
  enumerable: true,
  get: function () {
    return _ukUA.default;
  }
});
Object.defineProperty(exports, "zh", {
  enumerable: true,
  get: function () {
    return _zh.default;
  }
});
Object.defineProperty(exports, "zhTW", {
  enumerable: true,
  get: function () {
    return _zhTW.default;
  }
});
var _Calendar = _interopRequireDefault(require("./Date/Calendar"));
var _DatePickerModal = _interopRequireDefault(require("./Date/DatePickerModal"));
var _DatePickerModalContent = _interopRequireDefault(require("./Date/DatePickerModalContent"));
var _TimePickerModal = _interopRequireDefault(require("./Time/TimePickerModal"));
var _TimePicker = _interopRequireDefault(require("./Time/TimePicker"));
var _DatePickerInput = _interopRequireDefault(require("./Date/DatePickerInput"));
var _utils = require("./translations/utils");
var _ar = _interopRequireDefault(require("./translations/ar"));
var _ca = _interopRequireDefault(require("./translations/ca"));
var _da = _interopRequireDefault(require("./translations/da"));
var _de = _interopRequireDefault(require("./translations/de"));
var _en = _interopRequireDefault(require("./translations/en"));
var _enGB = _interopRequireDefault(require("./translations/enGB"));
var _es = _interopRequireDefault(require("./translations/es"));
var _fi = _interopRequireDefault(require("./translations/fi"));
var _fr = _interopRequireDefault(require("./translations/fr"));
var _he = _interopRequireDefault(require("./translations/he"));
var _hi = _interopRequireDefault(require("./translations/hi"));
var _it = _interopRequireDefault(require("./translations/it"));
var _ko = _interopRequireDefault(require("./translations/ko"));
var _nl = _interopRequireDefault(require("./translations/nl"));
var _pl = _interopRequireDefault(require("./translations/pl"));
var _pt = _interopRequireDefault(require("./translations/pt"));
var _tr = _interopRequireDefault(require("./translations/tr"));
var _zh = _interopRequireDefault(require("./translations/zh"));
var _zhTW = _interopRequireDefault(require("./translations/zhTW"));
var _cs = _interopRequireDefault(require("./translations/cs"));
var _el = _interopRequireDefault(require("./translations/el"));
var _ru = _interopRequireDefault(require("./translations/ru"));
var _ro = _interopRequireDefault(require("./translations/ro"));
var _id = _interopRequireDefault(require("./translations/id"));
var _ja = _interopRequireDefault(require("./translations/ja"));
var _th = _interopRequireDefault(require("./translations/th"));
var _ukUA = _interopRequireDefault(require("./translations/ukUA"));
var _noNO = _interopRequireDefault(require("./translations/noNO"));
var _sv = _interopRequireDefault(require("./translations/sv"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
